msos/StackWalker.cs

```diff
@@ -0,0 +1,372 @@
+
+        private const int MAXIMUM_WAIT_OBJECTS = 0x00000102;
+
+        #endregion
+
+        protected readonly ClrRuntime _runtime;
+
+        public StackWalkerStrategy(ClrRuntime runtime)
+        {
+            _runtime = runtime;
+        }
+
+
+        private List<byte[]> ReadFromMemmory(ulong startAddress, uint count)
+        {
+            List<byte[]> result = new List<byte[]>();
+            int sum = 0;
+            //TODO: Check if dfor can be inserted into the REadMemmory result (seems to be..)
+            for (int i = 0; i < count; i++)
+            {
+                byte[] readedBytes = new byte[IntPtr.Size];
+                if (_runtime.ReadMemory(startAddress, readedBytes, IntPtr.Size, out sum))
+                {
+                    result.Add(readedBytes);
+                }
+                else
+                {
+                    throw new Exception($"Accessing Unreadable memorry at {startAddress}");
+                }
+                //Advancing the pointer by 4 (32-bit system)
+                startAddress += (ulong)IntPtr.Size;
+            }
+            return result;
+        }
+
+        internal bool CheckMethod(UnifiedStackFrame frame, string key)
+        {
+            bool result = frame != null
+                && !String.IsNullOrEmpty(frame.Method)
+                && frame.Method != null && frame.Method.Contains(key);
+
+            return result;
+        }
+        
+        internal bool GetCriticalSectionBlockingObject(UnifiedStackFrame frame, out UnifiedBlockingObject blockingObject)
+        {
+            bool result = false;
+
+            if (frame.Handles != null && frame.Method.Contains(ENTER_CRITICAL_SECTION_FUNCTION_NAME))
+            {
+                blockingObject = GetCriticalSectionBlockingObject(frame);
+                result = blockingObject != null;
+            }
+            else
+            {
+                blockingObject = null;
+            }
+
+            return result;
+        }
+
+        internal bool SetFrameParameters(UnifiedStackFrame frame, ClrRuntime runtime)
+        {
+            bool waitCallFound = false;
+
+            if (waitCallFound = CheckMethod(frame, WAIT_FOR_SINGLE_OBJECTS_FUNCTION_NAME))
+            {
+                DealWithSingle(frame);
+            }
+            else if (waitCallFound = CheckMethod(frame, WAIT_FOR_MULTIPLE_OBJECTS_FUNCTION_NAME))
+            {
+                DealWithMultiple(frame);
+            }
+            else if (waitCallFound = CheckMethod(frame, ENTER_CRITICAL_SECTION_FUNCTION_NAME))
+            {
+                DealWithCriticalSection(frame);
+            }
+
+            return waitCallFound;
+        }
+
+        protected void EnrichUnifiedStackFrame(UnifiedStackFrame frame, ulong waitCount, ulong hPtr)
+        {
+            if (waitCount > MAXIMUM_WAIT_OBJECTS)
+            {
+                throw new InvalidOperationException($"waitCount exceeded MAXIMUM_WAIT_OBJECTS :{ waitCount }");
```

Comment:
I really wouldn't throw an exception in that case. Just ignore the frame, or read the first `MAXIMUM_WAIT_OBJECTS`, but it really doesn't make sense that if the frame is slightly corrupted that you'd just throw an exception and crash the analysis.